addons/sale/models/sale_order_line.py

```diff
@@ -312,21 +310,24 @@ class SaleOrderLine(models.Model):
         comodel_name='sale.order.line',
         compute='_compute_parent_id',
         index='btree_not_null',
+        copy=False,
         store=True,
+        precompute=True,
```

Comment:
Why do you need to precompute the field here?
It's not meant for fields based on ids linked to the current record(s).

```python
self.env['sale.order'].create({
    'order_line': [Command.create(section_values), Command.create(line_values)]
})
```
The second line will have either a falsy parent_id, or the compute will be called twice (dropping the point of the `precompute` attribute), which is probably the case here considering your changes to the `depends` of `_compute_parent_id`.